src/devices/Common/Iot/Device/Multiplexing/IOutputSegment.cs

```diff
@@ -0,0 +1,45 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Device.Gpio;
+using System.Threading;
+
+namespace Iot.Device.Multiplexing
+{
+    /// <summary>
+    /// Interface that abstracts multiplexing over a segment of outputs.
+    /// </summary>
+    public interface IOutputSegment : IDisposable
+    {
+        /// <summary>
+        /// Length of segment (number of outputs)
+        /// </summary>
+        int Length { get; }
+
+        /// <summary>
+        /// Writes a PinValue to a multiplexed output.
+        /// Does not perform a latch.
```

Comment:
NIT: Can we clarify here what does performing or not a latch means? It would be good since theoretically new classes might want to implement this in the future, and if this is something that should be applied to all implementations, it may require a more detailed explanation

Reply:
How's this?

```csharp
        /// <summary>
        /// Writes a PinValue to a virtual output.
        /// Does not display output.
        /// </summary>
        void Write(int output, PinValue value);
```